File: ui-automation/selenium-javascript/tests/ecommerce/homepageLoad.test.js
const { expect } = require('chai');
const BaseTest = require('../../utils/BaseTest');
const EcommercePage = require('../../pages/EcommercePage');

describe('Homepage Load Test', function() {
    let baseTest;
    let driver;
    let ecommercePage;

    beforeEach(async function() {
        baseTest = new BaseTest();
        driver = await baseTest.setUp();
        ecommercePage = new EcommercePage(driver);
    });

    afterEach(async function() {
        await baseTest.tearDown();
    });

    it('should navigate to ecommerce-playground and verify search field is visible', async function() {
        // Navigate to the ecommerce playground
        await ecommercePage.navigateToHomepage();
        
        // Verify the search field is visible
        const isSearchVisible = await ecommercePage.isSearchInputVisible();
        expect(isSearchVisible).to.be.true;
    });
});

File: ui-automation/selenium-java/src/test/java/tests/ecommerce/HomepageLoadTest.java
package tests.ecommerce;

import org.testng.annotations.Test;
import org.testng.Assert;
import pages.EcommercePage;
import utils.BaseTest;

/**
 * Test class for Homepage Load functionality
 * Mirrors the homepageLoad.lambda.cy.js Cypress test
 */
public class HomepageLoadTest extends BaseTest {
    
    @Test(description = "Should navigate to ecommerce-playground and verify search field is visible")
    public void shouldNavigateToEcommercePlayground() {
        EcommercePage ecommercePage = new EcommercePage();
        
        // Navigate to the ecommerce playground
        ecommercePage.navigateToHomepage();
        
        // Verify the search field is visible
        Assert.assertTrue(ecommercePage.isSearchInputVisible(), 
            "Search field should be visible on the homepage");
    }
}
